From 0471d76499aae8f44cb21c7faa1609e0387abafb Mon Sep 17 00:00:00 2001
From: Pepper Gray <hello@peppergray.xyz>
Date: Sat, 11 Apr 2026 15:08:26 +0200
Subject: [PATCH] expose alias for stat64 and lstat64 on musl

fix: 477
Signed-off-by: Pepper Gray <hello@peppergray.xyz>
---
 src/selinux_restorecon.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/selinux_restorecon.c b/src/selinux_restorecon.c
index 8425a3c5f..2ccda300e 100644
--- a/src/selinux_restorecon.c
+++ b/src/selinux_restorecon.c
@@ -19,7 +19,6 @@
 #include <limits.h>
 #include <stdint.h>
 #include <sys/types.h>
-#include <sys/stat.h>
 #include <sys/xattr.h>
 #include <sys/vfs.h>
 #include <sys/statvfs.h>
@@ -29,6 +28,10 @@
 #include <syslog.h>
 #include <assert.h>
 
+// expose alias for stat64 and lstat64 on musl
+#define _LARGEFILE64_SOURCE 1
+#include <sys/stat.h>
+
 #include <selinux/selinux.h>
 #include <selinux/context.h>
 #include <selinux/label.h>
